include/aws/common/invariants.h

```diff
@@ -0,0 +1,53 @@
+#ifndef AWS_COMMON_INVARIANTS_H
+#define AWS_COMMON_INVARIANTS_H
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/assert.inl>
+#include <aws/common/macros.h>
+
+/**
+ * Define function contracts.
+ * When the code is being verified using CBMC these contracts are formally verified;
+ * When the code is built in debug mode, they are checked as much as possible using assertions
+ * When the code is built in production mode, they are not checked.
+ * Violations of the function contracts are undefined behaviour.
+ */
+#ifdef CBMC
```

Comment:
I'd vote to merge this file into assert.h

Reply:
Done